src/hal/environment.cpp

```diff
@@ -129,48 +155,39 @@ void OswHal::Environment::setupStepStatistics() {
         if ((resetDay >= 0 && resetDay < 8) && (unsigned short) resetDay == currDoW + 1) // (e.g. 1 - 7 are days, 0 is disabled)
             this->_stepsSum = 0;
 
-        OSW_LOG_D("Updated steps from DoW ", lastDoW, " to DoW ", currDoW);
-        lastDoW = currDoW;
+        this->_stepsLastDoW = currDoW;
+        OSW_LOG_D("Updated steps from DoW ", this->_stepsLastDoW, " to DoW ", currDoW);
+
         // write step cache + stepsum
         res = prefs.putBytes(PREFS_STEPS_STATS, &this->_stepsCache, sizeof(this->_stepsCache)) == sizeof(this->_stepsCache);
         assert(res);
         res = prefs.putUInt(PREFS_STEPS_ALL, this->_stepsSum) == sizeof(this->_stepsSum);
         assert(res);
-        res = prefs.putUInt(PREFS_STEPS_DAYLASTCHECKED, lastDoW) == sizeof(lastDoW);
+        res = prefs.putUInt(PREFS_STEPS_DAYLASTCHECKED, this->_stepsLastDoW) == sizeof(this->_stepsLastDoW);
         assert(res);
+        prefs.end();
     }
-    prefs.end();
+
 #ifndef NDEBUG
```

Comment:
Yes, but I do not think it would be bad to give the compiler the clear hint (by removing the code), that he does not even need to construct the string at all...